### PR TITLE
Use Chat.html`` a bit more

### PR DESCRIPTION
**Pull request by urkerab**

### CODE REVIEW

File: server/chat-commands.js [hunk 1]
```diff
@@ -798,7 +798,7 @@ const commands = {
 
 		Ladders.visualizeAll(target).then(values => {
 			let buffer = `<div class="ladder"><table>`;
-			buffer += `<tr><td colspan="8">User: <strong>${Chat.escapeHTML(target)}</strong></td></tr>`;
+			buffer += Chat.html`<tr><td colspan="8">User: <strong>${target}</strong></td></tr>`;
```
Comment (TheJetOU) [inline server/chat-commands.js]: Won't this cause real HTML to get escaped?

Comment (Zarel) [reply server/chat-commands.js]: There's no real HTML in here...

File: server/chat-commands.js [hunk 2]
```diff
@@ -828,26 +828,26 @@ const commands = {
 		if (!id) return this.parse('/help makechatroom');
 		if (id.length > MAX_CHATROOM_ID_LENGTH) return this.errorReply("The given room title is too long.");
 		// Check if the name already exists as a room or alias
-		if (Rooms.search(id)) return this.errorReply(`The room '${target}' already exists.`);
-		if (!Rooms.global.addChatRoom(target)) return this.errorReply(`An error occurred while trying to create the room '${target}'.`);
+		if (Rooms.search(id)) return this.errorReply(Chat.html`The room '${target}' already exists.`);
```
Comment (Zarel) [inline server/chat-commands.js]: `errorReply` does not take HTML; this will lead to double-escaping.

Comment (urkerab) [reply server/chat-commands.js]: Ah, that goes for `sendReply` as well I take it. Thanks, that had confused me...

Comment (Zarel) [reply server/chat-commands.js]: `sendReply` sends a protocol message. If you do something like ``sendReply(`|html|...`)``, that will need escaping.

File: server/chat-plugins/mafia.js [hunk 3]
```diff
@@ -1044,7 +1044,7 @@ class MafiaTracker extends Rooms.RoomGame {
 			if (this.subs.includes(targetUser.id)) this.subs.splice(this.subs.indexOf(targetUser.id), 1);
 			this.played.push(targetUser.id);
 			this.playerTable[targetUser.id] = player;
-			this.sendRoom(`${Chat.escapeHTML(targetUser.name)} has been added to the game by ${Chat.escapeHTML(user.name)}!`, {declare: true});
+			this.sendRoom(Chat.html`${targetUser.name} has been added to the game by ${user.name}!`, {declare: true});
```
Comment (Zarel) [inline server/chat-plugins/mafia.js]: `{declare: boolean}` as an option? Ick.

File: server/chat-plugins/scavengers.js [hunk 4]
```diff
@@ -193,7 +193,7 @@ function formatQueue(queue = [], viewer, room, broadcasting) {
 			const queuedBy = item.hosts.every(h => h.id !== item.staffHostId) ? ` / ${item.staffHostId}` : '';
 			let questions;
 			if (!broadcasting && (item.hosts.some(h => h.id === viewer.id) || viewer.id === item.staffHostId)) {
-				questions = item.questions.map((q, i) => i % 2 ? `<span style="color: green"><em>[${Chat.escapeHTML(q.join(' / '))}]</em></span><br />` : Chat.escapeHTML(q)).join(" ");
+				questions = item.questions.map((q, i) => i % 2 ? Chat.html`<span style="color: green"><em>[${q.join(' / ')}]</em></span><br />` : Chat.escapeHTML(q)).join(" ");
```
Comment (Zarel) [inline server/chat-plugins/scavengers.js]:
```suggestion
				questions = item.questions.map((q, i) => i % 2 ? (
					Chat.html`<span style="color: green"><em>[${q.join(' / ')}]</em></span><br />`
				) : (
					Chat.escapeHTML(q)
				)).join(" ");
```

This is pretty unreadable without this.

File: server/chat-plugins/scavengers.js [hunk 5]
```diff
@@ -1034,7 +1034,7 @@ let commands = {
 				if (!players.length) {
 					str += `<tr><td>${questionNum}</td><td>None</td>`;
 				} else {
-					str += `<tr><td>${questionNum}</td><td>${players.map(pl => pl.lastGuess > Date.now() - 1000 * 300 ? `<strong>${Chat.escapeHTML(pl.name)}</strong>` : Chat.escapeHTML(pl.name)).join(", ")}`;
+					str += `<tr><td>${questionNum}</td><td>${players.map(pl => pl.lastGuess > Date.now() - 1000 * 300 ? Chat.html`<strong>${pl.name}</strong>` : Chat.escapeHTML(pl.name)).join(", ")}`;
```
Comment (Zarel) [inline server/chat-plugins/scavengers.js]:
```suggestion
					str += `<tr><td>${questionNum}</td><td>`;
					str += players.map(pl => pl.lastGuess > Date.now() - 1000 * 300 ?
						Chat.html`<strong>${pl.name}</strong>`
					:
						Chat.escapeHTML(pl.name)
					).join(", ");
```

(readability pass)

**Comment by Zarel** [comment]
Everything else looks good.

File: server/chat-plugins/mafia.js (hunk 3 above)
Comment (Zarel) [reply server/chat-plugins/mafia.js]: (This should be refactored into two different functions `sendRoom` and `declareRoom`, but that doesn't need to be in this PR.)

File: server/chat-commands.js (hunk 2 above)
Comment (urkerab) [reply server/chat-commands.js]: Right, but none of the cases I touched did that, which is why I reverted all of those changes.

File: server/chat-plugins/scavengers.js (hunk 4 above)
Comment (urkerab) [reply server/chat-plugins/scavengers.js]:
```
   198:1  error  Expected indentation of 5 tabs but found 4  indent
   198:6  error  Bad line breaking before and after ':'      operator-linebreak
```

File: server/chat-plugins/scavengers.js (hunk 5 above)
Comment (urkerab) [reply server/chat-plugins/scavengers.js]:
```
  1044:1  error  Expected indentation of 6 tabs but found 5  indent
  1044:7  error  Bad line breaking before and after ':'      operator-linebreak
```

File: server/chat-plugins/mafia.js (hunk 3 above)
Comment (urkerab) [reply server/chat-plugins/mafia.js]: It actually accepts 4 boolean options, `uhtml`, `declare`, `strong`, and `timestamp` (in order of priority).

File: server/chat-plugins/scavengers.js (hunk 4 above)
Comment (Zarel) [reply server/chat-plugins/scavengers.js]: Can you play around with it until it stops complaining? Maybe it needs parens?

Comment (Zarel) [reply server/chat-plugins/scavengers.js]:
```suggestion
				questions = item.questions.map((q, i) => (i % 2 ?
					Chat.html`<span style="color: green"><em>[${q.join(' / ')}]</em></span><br />`
				:
					Chat.escapeHTML(q)
				)).join(" ");
```

Try this?

Comment (urkerab) [reply server/chat-plugins/scavengers.js]: I looked into this and I don't think `eslint` allows a bare `:` at all; you either need to have it all on one line (if it fits), or you can (as currently configured in `.eslintrc`) only allow a newline after the `:` to break the line up, or the other setting would only allow a newline before the `:` to break the line up. (Unless you override `eslint` for this line of course.)

Comment (Zarel) [reply server/chat-plugins/scavengers.js]: Can you override `eslint` to permit it? Feel free to do that in a separate PR; I won't block this one on that.

**Comment by Zarel** [comment]
Sorry, this conflicted with the `chat-commands/` refactor. Can you rebase?

**Comment by Zarel** [comment]
I think you've fixed everything important, so I can merge immediately upon your rebase.

**Comment by urkerab** [comment]
Ugh, that rebase was hard.

**Pull request merged**